lib/mauth/client/authenticator_base.rb

```diff
@@ -22,7 +22,17 @@ def authentic?(object)
       # is set. Otherwise will authenticate with only the highest protocol version present
       def authenticate!(object)
         if object.protocol_version == 2
-          authenticate_v2!(object)
+          begin
+            authenticate_v2!(object)
+          rescue InauthenticError => e
+            raise e if v2_only_authenticate?
+
+            object.fall_back_to_mws_signature_info
+            raise e unless object.signature
+
+            log_authentication_request(object)
+            authenticate_v1!(object)
```

Comment:
Can you add a `warning` log after this line so we know that we hit this case? Something like `Completed successful authentication attempt after fallback to v1`

Reply:
added in 31d5b62

Reply:
Should there be two warnings : 1) When V2 authenticate fails  and fall back to authenticate for V1  2) When fall back V1 authenticate is passed 
Or just one as above 1) When V2 fails and failback to V1 is passed ??

Reply:
will let @mdsol/team-16 to decide but I don't think we need both

Reply:
I think one here makes more sense, when both fail it's just a plain old inauthentic request and these `authenticate!` methods take care of logging.